junit-platform-engine/src/main/java/org/junit/platform/engine/support/descriptor/MethodSource.java

```diff
@@ -167,6 +175,67 @@ public boolean equals(Object o) {
 				&& Objects.equals(this.methodParameterTypes, that.methodParameterTypes);
 	}
 
+	/**
+	 * Get the {@linkplain Class Java class} of this source.
+	 *
+	 * <p>If the {@link Class} was not provided, but only the name, this method
+	 * attempts to lazily load the {@link Class} based on its name and throws a
+	 * {@link PreconditionViolationException} if the class cannot be loaded.
+	 *
+	 * @since 5.7
+	 * @see #getClassName()
+	 */
+	public final Class<?> getJavaClass() {
+		if (this.javaClass == null) {
+			this.javaClass = ReflectionUtils.tryToLoadClass(this.className).getOrThrow(
+				cause -> new PreconditionViolationException("Could not load class with name: " + this.className,
+					cause));
+		}
+		return this.javaClass;
+	}
+
+	/**
+	 * Get the {@linkplain Method Java method} of this source and any overloads.
+	 *
+	 * <p>This method attempts to lazily load the {@link Method} list based on
+	 * the method's name and throws a {@link PreconditionViolationException} if
+	 * the methods cannot be loaded.
+	 *
+	 * <p>An empty list will be returned if there are no matching methods.
+	 *
+	 * @since 5.7
+	 * @see #getMethodName()
+	 * @see #getJavaMethod()
+	 */
+	public List<Method> getJavaMethods() {
+		if (this.javaMethods == null) {
+			this.javaMethods = ReflectionUtils.findMethods(getJavaClass(),
+				method -> method.getName().equals(methodName));
+		}
+		return this.javaMethods;
+	}
+
+	/**
+	 * Get the {@linkplain Method Java method} of this source.
+	 *
+	 * <p>If the {@link Method} was not provided, but only the name, this method
+	 * attempts to lazily load the {@link Method} based on its name and throws a
+	 * {@link PreconditionViolationException} if the method cannot be loaded or
+	 * if the method has overloads.
+	 *
+	 * @since 5.7
+	 * @see #getMethodName()
+	 * @see #getJavaMethods()
+	 */
+	public Method getJavaMethod() {
```

Comment:
Please implement this method analogous to `org.junit.platform.engine.discovery.MethodSelector.getJavaMethod()`.

Reply:
ok